Dungeoneer/src/com/interrupt/dungeoneer/statuseffects/BurningEffect.java

```diff
@@ -103,6 +103,177 @@ public void onStatusBegin(Actor owner) {
 	public void onStatusEnd(Actor owner) {
 		this.active = false;
 		Fire fire = (Fire)owner.getAttached(Fire.class);
-		if(fire != null) fire.isActive = false;
+		if (fire != null) fire.isActive = false;
 	}
+
+
+    /** Used for putting out the fire */
+	private boolean isEntityInWater() {
+        if (owner.inwater) {
+            Tile waterTile = Game.instance.level.findWaterTile(owner.x, owner.y, owner.z, owner.collision);
+            return waterTile.data != null && (waterTile.data.hurts <= 0 || waterTile.data.damageType != DamageType.FIRE);
```

Comment:
Simplified the if statement - deleted the "waterTile != null", because if the waterTile contains the data (!=null) then also the object shouldn't be null